src/openshift/component.ts

```diff
@@ -171,10 +180,11 @@ export class Component extends OpenShiftItem {
 
         const componentName = await vscode.window.showInputBox({
             prompt: "Component name",
-            validateInput: (value: string) => {
+            validateInput: async (value: string) => {
```

Comment:
Starting from this one all validations below are code duplication or CDs. Even if the message is different. Try to create a additional object  which would be Validator Factory and function to create isEmpty validator. Validator created would be a function that check if name is empty and return provided message. That would look like:
``` JavaScript
export const Validator = {
    emptyName: function (message: string): string | null | undefined {
        return function(name: string) {
            if(validator.isEmpty(name.trim()) {
                return message;
        }
    } 
}
```
so everywhere when you need to check for empty name you would use
``` JavaScript
Validator.emptyName('Message about something being empty');
```
To validate git repo URL is empty you would do something like this

``` JavaScript
        const repoURI = await vscode.window.showInputBox({
            prompt: 'Git repository URI',
            validateInput: (value: string) => {
                let message = Validator.emptyName('Empty Git repository URL')(value);
                if(message) {
                     return message;
                }
                if (!validator.isURL(value)) {
                     return 'Invalid URL provided';
                 }
             }
```